lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/CategoryDaoImpl.java

```diff
@@ -11,11 +13,14 @@
  *
  * @author Jakub Polacek
  */
+@Repository
+@Transactional
 public class CategoryDaoImpl implements CategoryDao {
 
-    @PersistenceContext(unitName = "category-unit", type = PersistenceContextType.EXTENDED)
+    @PersistenceContext
     private EntityManager em;
 
+    @Override
     public void addCategory(Category category) throws IllegalArgumentException {
         if (category == null) {
             throw new IllegalArgumentException("Category");
```

Comment:
Can you add an if for when category.getId() != null to throw an exception? Or is there a use case when we don't want to do that?

Reply:
fixol som ja